modules/internal/ChapelBase.chpl

```diff
@@ -675,9 +675,11 @@ module ChapelBase {
 
   // Removing the 'eltType' arg results in errors for --baseline
   inline proc _ddata_shift(type eltType, data: _ddata(eltType), shift: integral) {
-    var ret: _ddata(eltType);
-     __primitive("shift_base_pointer", ret, data, shift);
-    return ret;
+    local {
```

Comment:
I needed to add PRIM_ARRAY_SHIFT_BASE_POINTER to the switch in optmizeOnClauses's classifyPrimitive. I believe that this primitive is only ever used in a way that has its arguments be local. If one of the arguments were a wide_ref(ddata), or if the offset were a wide_ref(int), the primitive could conceivably generate communication. I added this local block here to emphasize that the use of the primitive should not be local.

So, I could change it in one of the following ways:
 * add a comment next to that local block
 * check for FLAG_WIDE_REF in either argument in classifyPrimitive
 * assert that FLAG_WIDE_REF is not set on the arguments in the CallExpr::codegen for this primitive

I'd be happy to do any of those just let me know your preference.